### PR TITLE
Minor tweaks to the .editorconfig and reformatted the entire code base

### PR DESCRIPTION
**Pull request by dennisdoomen**
The first commits change the `.editorconfig`. The final one used Rider and the "Safe Cleanup" profile to reformat.

### CODE REVIEW

File: Tests/FluentAssertions.Specs/Primitives/ObjectAssertionSpecs.cs
Comment (jnyrup) [inline Tests/FluentAssertions.Specs/Primitives/ObjectAssertionSpecs.cs:813]: Seems ReSharper's splitting algorithm has a [greedy](https://en.wikipedia.org/wiki/Greedy_algorithm) splitting approach? 

```cs
Action act = () => someObject.Should().NotBeAssignableTo(typeof(DummyImplementingClass), 
	"because we want to test the failure {0}", "message");
```

File: Tests/FluentAssertions.Specs/Types/TypeAssertionSpecs.HaveConstructor.cs
Comment (jnyrup) [inline Tests/FluentAssertions.Specs/Types/TypeAssertionSpecs.HaveConstructor.cs:26]: I liked the previous indentation as it makes it more clear that `HaveAccessModifier` binds to the constructor and not the type.

An alternative would be
```cs
.Which.Should().HaveAccessModifier(CSharpAccessModifier.Private);
```

File: Tests/FluentAssertions.Specs/Types/TypeExtensionsSpecs.cs
Comment (jnyrup) [inline Tests/FluentAssertions.Specs/Types/TypeExtensionsSpecs.cs:211]: Kind of liked this explicitness here and for `CompilerGenerated` below as we're trying to match _exactly_ what types are used in a `record struct`

File: Tests/FluentAssertions.Specs/Formatting/MultidimensionalArrayFormatterSpecs.cs
Comment (jnyrup) [inline Tests/FluentAssertions.Specs/Formatting/MultidimensionalArrayFormatterSpecs.cs:31]: This triggers

```
SA1500
Braces for multi-line statements should not share line
```

**Review by jnyrup — CHANGES_REQUESTED** [review]
Lot's of goodies 🚀 
Some I disagree with 🤔 
And a few that are wrong ❌

File: Src/FluentAssertions/Equivalency/Steps/StructuralEqualityEquivalencyStep.cs
Comment (dennisdoomen) [reply Src/FluentAssertions/Equivalency/Steps/StructuralEqualityEquivalencyStep.cs:103]: I agree with you. Let me check if a different setting works better.

File: Src/FluentAssertions/Execution/MessageBuilder.cs
Comment (dennisdoomen) [reply Src/FluentAssertions/Execution/MessageBuilder.cs:102]: It's an automatic format. In reality, we should have split the string.

File: Src/FluentAssertions/Equivalency/Steps/StructuralEqualityEquivalencyStep.cs
Comment (dennisdoomen) [reply Src/FluentAssertions/Equivalency/Steps/StructuralEqualityEquivalencyStep.cs:103]: It's a shame. It's an all or nothing thing. So either the auto-format will add the explicit type everywhere, or remove them. So we have three options:

1. Always remove them and override the individual cases
2. Always add them
3. Never touch what's there and leave it up to the developer to decide when the type is appropriate or not

File: Src/FluentAssertions/Formatting/DateTimeOffsetValueFormatter.cs
Comment (dennisdoomen) [reply Src/FluentAssertions/Formatting/DateTimeOffsetValueFormatter.cs:85]: Changed the settings to not even remove those explict `else` statements.

File: Src/FluentAssertions/Equivalency/Steps/StructuralEqualityEquivalencyStep.cs
Comment (dennisdoomen) [reply Src/FluentAssertions/Equivalency/Steps/StructuralEqualityEquivalencyStep.cs]: This seems to be doing the trick

```
resharper_object_creation_when_type_not_evident = explicitly_typed
resharper_object_creation_when_type_evident = target_typed
```

File: Src/FluentAssertions/Collections/GenericCollectionAssertions.cs
Comment (dennisdoomen) [reply Src/FluentAssertions/Collections/GenericCollectionAssertions.cs]: I like that idea, but it's outside of the scope of this PR. That's what I use 🏕️ or 📌 for.

File: Src/FluentAssertions/Equivalency/Steps/DataTableEquivalencyStep.cs
Comment (dennisdoomen) [reply Src/FluentAssertions/Equivalency/Steps/DataTableEquivalencyStep.cs]: I agree. Tweaking the settings.

File: Src/FluentAssertions/Equivalency/Steps/GenericDictionaryEquivalencyStep.cs
Comment (dennisdoomen) [reply Src/FluentAssertions/Equivalency/Steps/GenericDictionaryEquivalencyStep.cs]: I did some testing. It seems to be correct because it's a nested statement.

File: Tests/FluentAssertions.Specs/Types/PropertyInfoSelectorSpecs.cs
Comment (dennisdoomen) [reply Tests/FluentAssertions.Specs/Types/PropertyInfoSelectorSpecs.cs]: image

And for line 508:

image

File: Tests/FluentAssertions.Specs/Types/TypeExtensionsSpecs.cs
Comment (dennisdoomen) [reply Tests/FluentAssertions.Specs/Types/TypeExtensionsSpecs.cs]: You'll have to explicitly tell Rider/Resharper not to remove the unnecessary namespace specification.

File: Tests/FluentAssertions.Specs/Types/TypeAssertionSpecs.HaveConstructor.cs
Comment (dennisdoomen) [reply Tests/FluentAssertions.Specs/Types/TypeAssertionSpecs.HaveConstructor.cs]: That may be true, but technically the auto-indention is correct.

File: Tests/FluentAssertions.Specs/Primitives/ObjectAssertionSpecs.cs
Comment (dennisdoomen) [reply Tests/FluentAssertions.Specs/Primitives/ObjectAssertionSpecs.cs]: No idea what you're trying to say with that 🤷‍♂️

File: Src/FluentAssertions/Execution/AssertionScope.cs
Comment (dennisdoomen) [reply Src/FluentAssertions/Execution/AssertionScope.cs]: When did you get that one? Because my code compiled and all tests succeeded.

File: Src/FluentAssertions/Types/MethodInfoSelector.cs
Comment (dennisdoomen) [reply Src/FluentAssertions/Types/MethodInfoSelector.cs]: Nice trick!

File: Tests/FluentAssertions.Specs/AssertionOptionsSpecs.cs
Comment (dennisdoomen) [reply Tests/FluentAssertions.Specs/AssertionOptionsSpecs.cs]: In the updated PR, it no longer does that for simple initializers.

File: Tests/FluentAssertions.Specs/Collections/CollectionAssertionSpecs.BeInAscendingOrder.cs
Comment (dennisdoomen) [reply Tests/FluentAssertions.Specs/Collections/CollectionAssertionSpecs.BeInAscendingOrder.cs]: It's a simple rule. If it's longer than 130 characters it'll try to make it shorter. So our names shouldn't be that long.

File: Tests/FluentAssertions.Specs/Collections/CollectionAssertionSpecs.Equal.cs
Comment (dennisdoomen) [reply Tests/FluentAssertions.Specs/Collections/CollectionAssertionSpecs.Equal.cs]: The new PR keeps it as it was.

File: Tests/FluentAssertions.Specs/Primitives/ObjectAssertionSpecs.cs
Comment (jnyrup) [reply Tests/FluentAssertions.Specs/Primitives/ObjectAssertionSpecs.cs]: "Greedy" is a term in CS where an algorithm picks what immediately seems as the best choice, but in the larger picture turns out not to be optimal.

For this case I wanted to imply that splitting on `=>` and betwen `because` and `becauseArgs` is not optimal.
Following the the 130-rule the code, if we split after the type argument, we get two lines of instead of three lines ReSharper suggests.
Aesthetically I also prefer having `because` and `becauseArgs` closer.

File: Tests/FluentAssertions.Specs/Execution/AssertionScope.ContextDataSpecs.cs
Comment (dennisdoomen) [reply Tests/FluentAssertions.Specs/Execution/AssertionScope.ContextDataSpecs.cs]: No, the `#pragma` is no longer necessary. Whether the key doesn't exist at all or its value is `null`, `Get` will return `default(T)`

File: Src/FluentAssertions/Execution/AssertionScope.cs
Comment (jnyrup) [reply Src/FluentAssertions/Execution/AssertionScope.cs]: image

File: Tests/FluentAssertions.Specs/Primitives/ObjectAssertionSpecs.cs
Comment (dennisdoomen) [reply Tests/FluentAssertions.Specs/Primitives/ObjectAssertionSpecs.cs]: Me too, but that should be a case-by-case decision. The tool is not smart enough to think like us.

File: Src/FluentAssertions/Execution/AssertionScope.cs
Comment (dennisdoomen) [reply Src/FluentAssertions/Execution/AssertionScope.cs]: Ah, that only happens for me if I allow Rider to replace `new FailReason` with `new()`. But I've disabled that in my rework (which I haven't pushed yet).

File: Tests/FluentAssertions.Specs/Primitives/ObjectAssertionSpecs.cs
Comment (jnyrup) [reply Tests/FluentAssertions.Specs/Primitives/ObjectAssertionSpecs.cs]: For the aesthetics I agree it can't easily think like us, but if the objective is to do as few line splits to satisfy the 130-rule it's not impossible, it's just be more expensive to compute. (I'll stop the CS rambling now)

Comment (dennisdoomen) [reply Tests/FluentAssertions.Specs/Primitives/ObjectAssertionSpecs.cs]: After the merge, I'm going to make it a hard rule so that authors are forced to think about this.

File: Tests/FluentAssertions.Specs/Extensions/ObjectExtensionsSpecs.cs
Comment (dennisdoomen) [reply Tests/FluentAssertions.Specs/Extensions/ObjectExtensionsSpecs.cs]: I like that too, but I couldn't find a setting that would trigger Rider/R# to format it as such. And aligning as you had in your original example looks ugly on my machine, since I prefer to use a proportional font.

File: Tests/FluentAssertions.Specs/Types/PropertyInfoAssertionSpecs.cs
Comment (dennisdoomen) [reply Tests/FluentAssertions.Specs/Types/PropertyInfoAssertionSpecs.cs:153]: I have not found a way to change the behavior, so we'll have to live with that for auto-formatted files.

File: Tests/FluentAssertions.Specs/Execution/AssertionScope.ContextDataSpecs.cs
Comment (dennisdoomen) [reply Tests/FluentAssertions.Specs/Execution/AssertionScope.ContextDataSpecs.cs:44]: I wonder why Rider doesn't complain about this. It's using `dotnet build` under the hood.

File: Tests/FluentAssertions.Specs/Events/EventAssertionSpecs.cs
Comment (dennisdoomen) [reply Tests/FluentAssertions.Specs/Events/EventAssertionSpecs.cs:733]: Had to use `new\(\)\n.*\{.*\}`, but found them all ;-)

File: Src/FluentAssertions/Formatting/TimeSpanValueFormatter.cs
Comment (dennisdoomen) [reply Src/FluentAssertions/Formatting/TimeSpanValueFormatter.cs:79]: That's what I use 🏕️ for ;-)

File: Src/FluentAssertions/Common/ExpressionExtensions.cs
Comment (dennisdoomen) [reply Src/FluentAssertions/Common/ExpressionExtensions.cs:159]: So let's leave it for now.

File: Tests/FluentAssertions.Specs/Types/TypeExtensionsSpecs.cs
Comment (dennisdoomen) [reply Tests/FluentAssertions.Specs/Types/TypeExtensionsSpecs.cs:6]: I don't get this one. Is there a problem with the change?

Comment (jnyrup) [reply Tests/FluentAssertions.Specs/Types/TypeExtensionsSpecs.cs:6]: Another place in this file it seems I managed to persuade you to use fully qualified named for `StringBuilder` I recall it was.

File: Tests/FluentAssertions.Specs/Execution/AssertionScope.ContextDataSpecs.cs
Comment (jnyrup) [inline Tests/FluentAssertions.Specs/Execution/AssertionScope.ContextDataSpecs.cs:46]: Not at my dev machine right now, but I recall this triggers [IDE0055](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0055) about `#pragma` being indented.

Alternatively assign the `null` to a variable and we can git rid of both `#pragma` and `// ReSharper`
```
int? someValue = null;
scope.AddNonReportable("SomeKey", someValue);
```

File: Tests/FluentAssertions.Specs/Collections/GenericDictionaryAssertionSpecs.cs
Comment (jnyrup) [inline Tests/FluentAssertions.Specs/Collections/GenericDictionaryAssertionSpecs.cs:2757]:
```suggestion
                new Dictionary<int, int> { [1] = 42 },
                new TrueReadOnlyDictionary<int, int>(new Dictionary<int, int> { [1] = 42 }),
```

File: Tests/FluentAssertions.Equivalency.Specs/CollectionSpecs.cs
Comment (jnyrup) [inline Tests/FluentAssertions.Equivalency.Specs/CollectionSpecs.cs:205]:
```suggestion
            new MyClass { InterfaceProperty = 1, ClassProperty = 42 }
```

File: Tests/FluentAssertions.Equivalency.Specs/CollectionSpecs.cs
Comment (jnyrup) [inline Tests/FluentAssertions.Equivalency.Specs/CollectionSpecs.cs:211]:
```suggestion
            new MyClass { InterfaceProperty = 1, ClassProperty = 1337 }
```

File: Tests/FluentAssertions.Equivalency.Specs/CollectionSpecs.cs
Comment (jnyrup) [inline Tests/FluentAssertions.Equivalency.Specs/CollectionSpecs.cs:2637]:
```suggestion
        var genericCollectionA = new List<ExceptionThrowingClass> { new() };

        var genericCollectionB = new List<ExceptionThrowingClass> { new() };
```

File: Tests/FluentAssertions.Specs/Formatting/FormatterSpecs.cs
Comment (jnyrup) [inline Tests/FluentAssertions.Specs/Formatting/FormatterSpecs.cs:934]:
```cs
var values = new CustomClass[]
{
    new() { IntProperty = 1 },
    new() { IntProperty = 2 }
};
```

**Review by jnyrup — APPROVED** [review]

**Pull request merged**
